src/all.rs

```diff
@@ -57,6 +125,58 @@ impl<'a, Trait: ?Sized + TraitQuery> Iterator for ReadTableTraitsIter<'a, Trait>
     }
 }
 
+impl<'a, Trait: ?Sized + TraitQuery> Iterator for AddedReadTableTraitsIter<'a, Trait> {
+    type Item = &'a Trait;
+    fn next(&mut self) -> Option<Self::Item> {
+        // Iterate the remaining table components that are registered,
+        // until we find one that exists in the table.
+        let (column, meta) = unsafe { zip_exact(&mut self.components, &mut self.meta) }
+            .find_map(|(&component, meta)| self.table.get_column(component).zip(Some(meta)))?;
+
+        // SAFETY: we know that the table row is a valid index???
```

Comment:
To be more explicit about this invariant, it would be good to add a safety comment to the `self.table_row` field. Something like ``// SAFETY: must correspond to a valid row in `self.table` ``.